File: app/src/main/java/com/blaxsoftware/directcallwidget/MainActivity.java
package com.blaxsoftware.directcallwidget;

import android.appwidget.AppWidgetManager;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.blaxsoftware.directcallwidget.appwidget.DirectCallWidgetProvider;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Intent updateWidget = new Intent(this, DirectCallWidgetProvider.class);
        updateWidget.setAction(AppWidgetManager.ACTION_APPWIDGET_UPDATE);
        sendBroadcast(updateWidget);
    }
}

File: app/src/main/kotlin/com/blaxsoftware/directcallwidget/ui/MainActivity.kt
/*
 * Direct Call Widget - The widget that makes contacts accessible
 * Copyright (C) 2025 Fer P. A.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.blaxsoftware.directcallwidget.ui

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.ui.Modifier
import com.blaxsoftware.directcallwidget.DirectCallWidgetApp
import dagger.hilt.android.AndroidEntryPoint
import dev.ferp.dcw.core.ui.theme.DirectCallWidgetTheme

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val appVersionName = (application as DirectCallWidgetApp).appVersionName
        setContent {
            DirectCallWidgetTheme {
                MainNavHost(
                    modifier = Modifier.fillMaxSize(),
                    startDestination = Settings(appVersionName),
                    appVersionName = appVersionName
                )
            }
        }
    }
}